pulsar-io/alluxio/src/main/java/org/apache/pulsar/io/alluxio/sink/AlluxioAbstractSink.java

```diff
@@ -0,0 +1,236 @@
+        alluxioSinkConfig = AlluxioSinkConfig.load(config);
+        alluxioSinkConfig.validate();
+
+        // initialize FileSystem
+        String alluxioMasterHost = alluxioSinkConfig.getAlluxioMasterHost();
+        int alluxioMasterPort = alluxioSinkConfig.getAlluxioMasterPort();
+        Configuration.set(PropertyKey.MASTER_HOSTNAME, alluxioMasterHost);
+        Configuration.set(PropertyKey.MASTER_RPC_PORT, alluxioMasterPort);
+        if (alluxioSinkConfig.getSecurityLoginUser() != null) {
+            Configuration.set(PropertyKey.SECURITY_LOGIN_USERNAME, alluxioSinkConfig.getSecurityLoginUser());
+        }
+        fileSystem = FileSystem.Factory.get();
+
+        // initialize alluxio dirs
+        String alluxioDir = alluxioSinkConfig.getAlluxioDir();
+        fileDirPath = alluxioDir.startsWith("/") ? alluxioDir : "/" + alluxioDir;
+        tmpFileDirPath = fileDirPath + "/tmp";
+
+        AlluxioURI alluxioDirPath = new AlluxioURI(fileDirPath);
+        if (!fileSystem.exists(alluxioDirPath)) {
+            fileSystem.createDirectory(alluxioDirPath);
+        }
+
+        AlluxioURI tmpAlluxioDirPath = new AlluxioURI(tmpFileDirPath);
+        if (!fileSystem.exists(tmpAlluxioDirPath)) {
+            fileSystem.createDirectory(tmpAlluxioDirPath);
+        }
+
+        createFileOptions = CreateFileOptions.defaults();
+
+        recordsNum = 0;
+        recordsToAck = Lists.newArrayList();
+        tmpFilePath = "";
+        alluxioState = AlluxioState.WRITE_STARTED;
+
+        lastRotationTime = System.currentTimeMillis();
+        rotationRecordsNum = alluxioSinkConfig.getRotationRecords();
+        rotationInterval =  alluxioSinkConfig.getRotationInterval();
+    }
+
+    @Override
+    public void write(Record<V> record) {
+        long now = System.currentTimeMillis();
+
+        switch (alluxioState.ordinal()) {
+            case 0:
+                try {
+                    writeToAlluxio(record);
+                    if (!shouldRotate(now)) {
+                        break;
+                    }
+                    alluxioState = AlluxioState.FILE_ROTATED;
+                } catch (AlluxioException | IOException e) {
+                    log.error("Unable to write record to alluxio.", e);
+                    record.fail();
+                }
+            case 1:
+                try {
+                    closeAndCommitTmpFile();
+                    alluxioState = AlluxioState.FILE_COMMITTED;
+                    ackRecords();
+                } catch (AlluxioException | IOException e) {
+                    log.error("Unable to flush records to alluxio.", e);
+                    failRecords();
+                }
+            case 2:
+                alluxioState = AlluxioState.WRITE_STARTED;
+                break;
+            default:
+                log.error("{} is not a valid state when writing record to alluxio temp dir {}.",
+                    alluxioState, tmpFileDirPath);
```

Comment:
nit: a good practice is to add `break` for default branch as well

Reply:
done